dynamiqs/options.py

```diff
@@ -1,11 +1,21 @@
 from __future__ import annotations
 
+from typing import Callable
+
 import equinox as eqx
-from jaxtyping import Scalar
+import jax
+from jax import Array
+from jaxtyping import PyTree, Scalar
 
 
 class Options(eqx.Module):
     save_states: bool = True
     verbose: bool = True
     cartesian_batching: bool = True
     t0: Scalar | None = None  # defaults to tsave[0]
+    save_extra: Callable[[Array], PyTree] | None = None
+
+    def __post_init__(self):
+        if self.save_extra is not None:
+            # use `jax.tree_util.Partial` to make `save_extra` a valid Pytree
+            self.save_extra = jax.tree_util.Partial(self.save_extra)
```

Comment:
Is it required that it is a PyTree here? I would argue not.

Reply:
Yes, because it's used in jitted code. I get a
```
TypeError: Value <function <lambda> at 0x1235cb5b0> with type <class 'function'> is not a valid JAX type
```
otherwise.